template/.pre-commit-config.yaml.j2

```diff
@@ -18,6 +18,7 @@ repos:
     rev: eeee35a89e69d5772bdee97db1a6a898467b686e # 1.0
     hooks:
       - id: fmt
+        entry: cargo +nightly fmt
```

Comment:
Is this what you meant @nightkr?

It has the downside that users need to understand and install this
`error: toolchain 'nightly-2025-01-15-x86_64-unknown-linux-gnu' is not installed`
This also applies to CI and such

I'm not sure if this complexity is worth it, seeing kube only using `+nightly`

```suggestion
        entry: cargo +nightly-2025-01-15 fmt
```

Reply:
Looks like `RUSTUP_TOOLCHAIN=beta cargo fmt` _will_ automatically install it if missing.

Reply:
![image](https://github.com/user-attachments/assets/95b774a6-deb7-45c1-ae2a-bb83dcaaa5ec)

Reply:
Can confirm, `RUSTUP_TOOLCHAIN=nightly-2025-01-15 cargo fmt` works for me :partying_face: 

Should we use that for pre-commit?
So `entry: RUSTUP_TOOLCHAIN=nightly-2025-01-15 cargo fmt`

Reply:
I'd say go for it.